File: 01-Languages/Dart/18-Testing/lib/calculator.dart
class Calculator {
  int add(int a, int b) => a + b;

  double divide(double a, double b) {
    if (b == 0) throw ArgumentError('division by zero');
    return a / b;
  }

  bool isPalindrome(String s) {
    var cleaned = s.toLowerCase().replaceAll(RegExp(r'[^a-z0-9]'), '');
    return cleaned == cleaned.split('').reversed.join('');
  }
}

File: 01-Languages/Kotlin/18-Testing/src/Calculator.kt
class Calculator {
    fun add(a: Int, b: Int): Int = a + b

    fun divide(a: Double, b: Double): Double {
        if (b == 0.0) throw IllegalArgumentException("division by zero")
        return a / b
    }

    fun isPalindrome(s: String): Boolean {
        val cleaned = s.filter { it.isLetterOrDigit() }.lowercase()
        return cleaned == cleaned.reversed()
    }
}
